src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Unix.cs

```diff
@@ -130,26 +130,11 @@ internal void ReplaceHandleIfNecessaryAfterFailedConnect()
 
         internal SocketError ReplaceHandle()
         {
-            // Copy out values from key options. The copied values should be kept in sync with the
-            // handling in SafeSocketHandle.TrackOption.  Note that we copy these values out first, before
-            // we change _handle, so that we can use the helpers on Socket which internally access _handle.
-            // Then once _handle is switched to the new one, we can call the setters to propagate the retrieved
-            // values back out to the new underlying socket.
-            bool broadcast = false, dontFragment = false, noDelay = false;
-            int receiveSize = -1, receiveTimeout = -1, sendSize = -1, sendTimeout = -1;
-            short ttl = -1;
-            LingerOption? linger = null;
-            if (_handle.IsTrackedOption(TrackedSocketOptions.DontFragment)) dontFragment = DontFragment;
-            if (_handle.IsTrackedOption(TrackedSocketOptions.EnableBroadcast)) broadcast = EnableBroadcast;
-            if (_handle.IsTrackedOption(TrackedSocketOptions.LingerState)) linger = LingerState;
-            if (_handle.IsTrackedOption(TrackedSocketOptions.NoDelay)) noDelay = NoDelay;
-            if (_handle.IsTrackedOption(TrackedSocketOptions.ReceiveBufferSize)) receiveSize = ReceiveBufferSize;
-            if (_handle.IsTrackedOption(TrackedSocketOptions.ReceiveTimeout)) receiveTimeout = ReceiveTimeout;
-            if (_handle.IsTrackedOption(TrackedSocketOptions.SendBufferSize)) sendSize = SendBufferSize;
-            if (_handle.IsTrackedOption(TrackedSocketOptions.SendTimeout)) sendTimeout = SendTimeout;
-            if (_handle.IsTrackedOption(TrackedSocketOptions.Ttl)) ttl = Ttl;
-
-            // Then replace the handle with a new one
+            // Collect values of trackable socket options marked by SafeSocketHandle.TrackSocketOption().
+            Span<int> optionValues = stackalloc int[SafeSocketHandle.TrackableOptionCount];
```

Comment:
Possible trap here: the options span can be uninitialized stack space since we're using `SkipLocalsInit`.
I think it's still correct as-is since your source of truth are the `_trackedOptions`, but we could add a comment about why it's fine / just always clear this.